graphs/possible_bipartition.py

```diff
@@ -5,8 +5,57 @@ def possible_bipartition(dislikes):
     """ Will return True or False if the given graph
         can be bipartitioned without neighboring nodes put
         into the same partition.
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(n)
+        Space Complexity: O(n)
     """
-    pass
+    
+    # Ensure there are dislikes to begin with
+    if not dislikes:
+        return True
 
+    # Set empty deck, visited set, and two teams
+    deck = deque()
+    visited = set()
+    left = set()
+    right = set()
+
+    # # If 0 has no dislikes, start with 1
+    if dislikes[0]:
+        start = 0
+    else:
+        start = 1
```

Comment:
What if there are multiple dogs at the beginning of `dislikes` that don't dislike any other dogs (are disconnected from the rest of the graph)? How might you account for this edge case?